modules/virt-virtctl-commands.adoc

```diff
@@ -29,70 +29,206 @@ To view a list of global command options that you can use with any `virtctl` com
+
+If a port number is not specified, the proxy runs on a random port.
+|===
+
+[id='exporting-vms_{context}']
+== Exporting VM volumes
+
+You can use `virtctl vmexport` commands to create, download, or delete a volume exported from a VM, VM snapshot, or persistent volume claim (PVC).
+
+.`virtctl vmexport` commands
+[width="100%",cols="1a,2a",options="header"]
+|===
+|Command |Description
+|`virtctl vmexport create <vmexport_name> --vm\|snapshot\|pvc=<object_name>`
+|Create a `VirtualMachineExport` custom resource (CR) to export a volume from a VM, VM snapshot, or PVC.
+
+* `--vm`: Exports the PVCs and memory dump of a VM.
+* `--snapshot`: Exports the PVCs contained in a `VirtualMachineSnapshot` CR.
+* `--pvc`: Exports a PVC.
+* Optional: `--ttl=1h` specifies the time to live. The default duration is 2 hours.
+
+|`virtctl vmexport delete <vmexport_name>`
+|Delete a `VirtualMachineExport` CR manually.
+
+|`virtctl vmexport download <vmexport_name> --output=<output_file> --volume=<volume_name>`
+|Download the volumes defined in a `VirtualMachineExport` CR.
+
+* `--output` specifies the file format. Example: `disk.img.gz`.
+* `--volume` specifies the volume to download. This flag is optional if only one volume is available.
+
+Optional:
+
+* `--keep-vme` retains the `VirtualMachineExport` CR after download. The default behavior is to delete the `VirtualMachineExport` CR after download.
+* `--insecure` enables an insecure HTTP connection.
+
+|`virtctl vmexport download <vmexport_name> --<vm\|snapshot\|pvc>=<object_name> --output=<output_file>`
+|Create a `VirtualMachineExport` CR and then download the volumes defined in the CR.
+|===
+
+[id='vm-memory-dump_{context}']
+== VM memory dump
+
+You can use the `virtctl memory-dump` command to output a virtual machine (VM) memory dump on a PVC. You can specify an existing PVC or use the `--create-claim` flag to create a new PVC.
+
+.Prerequisites
 
-|`virtctl vnc --kubeconfig=$KUBECONFIG --proxy-only=true <vmi-name>`
-|Display the port number and connect manually to the virtual machine instance by using any viewer through the VNC connection.
+* The PVC volume mode must be `FileSystem`.
+* The PVC must be large enough to contain the memory dump.
++
+The formula for calculating the PVC size is `(VMMemorySize + 100Mi) * FileSystemOverhead`, where `100Mi` is the memory dump overhead.
 
-|`virtctl vnc --kubeconfig=$KUBECONFIG --port=<port-number> <vmi-name>`
-|Specify a port number to run the proxy on the specified port, if that port is available. If a port number is not specified, the proxy runs on a random port.
+* You must enable the hot plug feature gate in the `HyperConverged` custom resource by running the following command:
++
+[source,terminal]
+----
+$ oc patch hco kubevirt-hyperconverged -n openshift-cnv \
+  --type json -p '[{"op": "add", "path": "/spec/featureGates", \
+  "value": "HotplugVolumes"}]'
+----
+
+.Downloading the memory dump
+
+// Workaround for 4.12
```

Comment:
I dont think it should state 'workaround' its not a workaround. for 4.13 I added an enhancement of adding a simpler call to the virtcl memory dump download which actually uses the vmexport to download the memory dump. using the vmexport is the basic way of doing it.